### ISSUE TITLE
Add challenges for characters and their pronunciation

### ISSUE THREAD

**Issue by adnan360** [LibreLingo/LibreLingo#1143]
**Is your feature request related to a problem? Please describe.**

Sometimes seeing letters on screens is not enough to learn how to write them on paper. Writing is an important part of learning a language and without such guidelines, learning is not complete. Right now there's no way to show how to write a letter.

Actually the [idea](https://github.com/kantord/LibreLingo/pull/1121#issuecomment-786842864) came from @kantor and I'm just opening an issue for it:
> I don't know the Bengali script, so there's no way for me to know how to pronounce or read the words that I see. LibreLingo should add some way to teach the characters in Bengali script, and how to combine those characters together to form words/sentences.

And possibly there should be challenges to test what they've learned.

**Describe the solution you'd like**
@kantor describes it better:
> ...One of the main idea that I have for solving this is adding another section along with Words and Phrases. It would be something like Characters. And it would generate challenges similar to the ones for words, but it would be only about characters and their pronunciation

Previous quote talks about "teaching" and this one about "challenges". So there's 2 aspects to it.

I have prepared a [letter handwriting animation](https://github.com/kantord/LibreLingo/pull/1121#issuecomment-789793975) code to aid with the teaching part. Maybe it can help in the process or not (feel free to discuss).

**Describe alternatives you've considered**
I've thought about video recording a handwriting gesture. But it can't be done by everyone or won't be of same quality and it would need more bandwidth to serve.

**Additional context**
Well, since we're talking about challenges here, how would we collect a correct answer? Through drawing? If so, we'd need OCR or something like that to validate input. Or is there other simple ways? Feel free to discuss possibilities.

**Comment by kantord** [comment]
> Well, since we're talking about challenges here, how would we collect a correct answer? Through drawing? If so, we'd need OCR or something like that to validate input. Or is there other simple ways? Feel free to discuss possibilities.

As a first, super-simple idea I can imagine simply a version of the listening challenges. You just have to listen to an audio and select out of a list, which character you heard. In the case of Bengali, maybe this could be a combination of 2 characters, so you'd basically hear a syllable and choose with character combination it belongs to.

The reverse of that could also be possible, basically seeing a character and choosing which sound matches it.

**Comment by kantord** [comment]
I'm gonna pin this, it should not be terribly hard to solve so maybe someone is interested in implementing it

**Comment by pur80a** [comment]
As I imagine it, a minimal character `skill.yaml` could look like this: 
```
Skill:
   Name: Alphabet   
   Id: alphabet000
   Thumbnails:
     - abc1
 
New Characters:
   - Character: α
     Transliteration: a
```

Which files must I change in order to get options challenges for such a character class? 

- add `Character` class in `apps/librelingo_types/librelingo_types/data_types.py`
- add `Character` class to `Skill` class in `apps/librelingo_types/librelingo_types/data_types.py`
- import `Character` class in `/apps/librelingo_yaml_loader/librelingo_yaml_loader/yaml_loader.py`
- add `_convert_character()` and `_convert_characters()` in `apps/librelingo_yaml_loader/librelingo_yaml_loader/yaml_loader.py`

Where is defined which class uses which challenges as in the [docs](https://librelingo.app/docs/courses/skill.html#how-skills-are-taught-to-the-user)?

**Comment by Skivling** [comment]
Transliteration definition would be great for typo detection if a macron or diacretic is missing but it's the same base letter, e.g. 'a' when it's 'ā'.

**Comment by kantord** [comment]
> Transliteration definition would be great for typo detection if a macron or diacretic is missing but it's the same base letter, e.g. 'a' when it's 'ā'.

for this, some kind or normalization could be used, I don't know if it can be modeled in such a simple way though. I mean for some languages it might be simple (just remove accents) where in others it might have complicated rules that involve multiple characters, etc

**Comment by Skivling** [comment]
True. This could get complicated in some languages very easily, although I think a basic version for the easy ones is worth implementing first.

**Comment by Skivling** [comment]
Draft PR implementing this in the Python scripts ^^
